pkg/overlay/overlay.go

```diff
@@ -105,8 +105,7 @@ func mountHelper(contentDir, source, dest string, _, _ int, graphOptions []strin
 				return mount, err
 			}
 		}
-
-		overlayOptions = fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s,private", source, upperDir, workDir)
+		overlayOptions = fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s,private", escapeColon(source), upperDir, workDir)
```

Comment:
Not scoped to this PR: but since we have a boilerplate for escaping we could also support escaping source and dest for other types not just overlay. No actionable on this comment just leaving here for note.